apple/internal/apple_framework_import.bzl

```diff
@@ -318,11 +318,36 @@ def _debug_info_binaries(
 
     return all_binaries_dict.values()
 
+def _get_framework_imports(ctx):
+    if not ctx.attr.xcframework_platform_ids:
+        return ctx.files.framework_imports
+    curr_platform = ctx.fragments.apple.single_arch_platform.name_in_plist.lower()
+    xcframework_path = ""
+    for f in ctx.files.framework_imports:
+        if ".xcframework" in f.path:
+            xcframework_path = f.path.split(".xcframework")[0] + ".xcframework"
+            break
+    if not xcframework_path.endswith(".xcframework"):
+        fail("couldn't find xcframework at framework_imports")
+    framework_name = paths.split_extension(paths.basename(xcframework_path))[0]
+
+    if curr_platform not in [p.lower() for p in ctx.attr.xcframework_platform_ids]:
+        fail(
+            "Missing framework path mapping for platform `{}`; is this platform supported?"
+                .format(str(curr_platform)),
+        )
+    else:
+        platform_path = "{}/{}/{}.framework".format(xcframework_path, ctx.attr.xcframework_platform_ids[curr_platform], framework_name)
+        framework_imports_for_platform = [f for f in ctx.files.framework_imports if platform_path in f.short_path]
```

Comment:
Might want `path` here over `short_path`, which will return a relative path when the framework imports are in a different repo.